.circleci/config.yml

```diff
@@ -1,17 +1,55 @@
+defaults: &defaults
+  docker:
+    - image: cryptape/cita-build:latest
+  working_directory: ~/cita-build
+
 version: 2
 jobs:
-  build:
-    docker:
-      - image: cryptape/cita-build:latest
-    working_directory: ~/cita-build
+  format:
+    <<: *defaults
     steps:
       - checkout
       - run:
           name: format
           command: 'cargo fmt --all -- --write-mode diff'
+  build:
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          key: build-cache
       - run:
           name: build
           command: '.circleci/loop_crates_to_run build'
+      - save_cache:
+          key: build-cache
```

Comment:
maybe we should do `cargo generate-lockfile` first,
then generate cache key like `build-deps-{{ checksum "Cargo.lock" }}` .

`cargo build` will not clean old deps, the cache use constant key will increase infinity.

Reply:
Agree with @zhangsoledad .

Reply:
Done.